src/serac/physics/solid_mechanics_contact.hpp

```diff
@@ -246,10 +241,135 @@ class SolidMechanicsContact<order, dim, Parameters<parameter_space...>,
+   */
+  void warmStartDisplacementContact(double dt)
+  {
+    SERAC_MARK_FUNCTION;
+
+    du_ = 0.0;
+    for (auto& bc : bcs_.essentials()) {
+      // apply the future boundary conditions, but use the most recent Jacobians stiffness.
+      bc.setDofs(du_, time_ + dt);
+    }
+
+    auto& constrained_dofs = bcs_.allEssentialTrueDofs();
+    for (int i = 0; i < constrained_dofs.Size(); i++) {
+      int j = constrained_dofs[i];
+      du_[j] -= displacement_(j);
+    }
+
+    if (use_warm_start_) {
+      // use the most recently evaluated Jacobian
+      auto [_, drdu] = (*residual_)(time_, shape_displacement_, differentiate_wrt(displacement_), acceleration_,
+                                    *parameters_[parameter_indices].previous_state...);
+      J_.reset();
+      J_ = assemble(drdu);
+
+      contact_.update(cycle_, time_, dt);
+      if (contact_.haveLagrangeMultipliers()) {
+        J_offsets_    = mfem::Array<int>({0, displacement_.Size(), displacement_.Size() + contact_.numPressureDofs()});
+        J_constraint_ = contact_.jacobianFunction(J_.release());
+
+        // take ownership of blocks
+        J_constraint_->owns_blocks = false;
+        J_ = std::unique_ptr<mfem::HypreParMatrix>(static_cast<mfem::HypreParMatrix*>(&J_constraint_->GetBlock(0, 0)));
+        J_12_ =
+            std::unique_ptr<mfem::HypreParMatrix>(static_cast<mfem::HypreParMatrix*>(&J_constraint_->GetBlock(0, 1)));
+        J_21_ =
+            std::unique_ptr<mfem::HypreParMatrix>(static_cast<mfem::HypreParMatrix*>(&J_constraint_->GetBlock(1, 0)));
+        J_22_ =
+            std::unique_ptr<mfem::HypreParMatrix>(static_cast<mfem::HypreParMatrix*>(&J_constraint_->GetBlock(1, 1)));
+
+        J_e_.reset();
+        J_e_    = bcs_.eliminateAllEssentialDofsFromMatrix(*J_);
+        J_e_21_ = std::unique_ptr<mfem::HypreParMatrix>(J_21_->EliminateCols(bcs_.allEssentialTrueDofs()));
+        J_12_->EliminateRows(bcs_.allEssentialTrueDofs());
+
+        J_operator_ = J_constraint_.get();
+      } else {
+        // get 11-block holding jacobian contributions
+        auto block_J         = contact_.jacobianFunction(J_.release());
+        block_J->owns_blocks = false;
+        J_ = std::unique_ptr<mfem::HypreParMatrix>(static_cast<mfem::HypreParMatrix*>(&block_J->GetBlock(0, 0)));
+
+        J_e_ = bcs_.eliminateAllEssentialDofsFromMatrix(*J_);
+
+        J_operator_ = J_.get();
+      }
+
+      // Update the linearized Jacobian matrix
+      mfem::Vector augmented_residual(displacement_.space().TrueVSize() + contact_.numPressureDofs());
+      augmented_residual     = 0.0;
+      const mfem::Vector res = (*residual_)(time_ + dt, shape_displacement_, displacement_, acceleration_,
+                                            *parameters_[parameter_indices].state...);
+
+      // TODO this copy is required as the sundials solvers do not allow move assignments because of their memory
```

Comment:
Is this a TODO or a NOTE?

Reply:
Probably a TODO once mfem/mfem#3531 is addressed, but it doesn't look like that will happen in the near term.  I'll change it to a NOTE